core/blockchain.go

```diff
@@ -483,6 +484,7 @@ func (bc *BlockChain) maintainTxIndex() {
 	if head := bc.CurrentBlock(); head != nil && head.Number.Uint64() > txLookupLimit {
 		done = make(chan struct{})
 		tail := rawdb.ReadTxIndexTail(bc.db)
+		bc.wg.Add(1)
```

Comment:
alternatively I think we can check `if done != nil { ... }` like its done in `bc.quit`, but also `wg` seems fine.